app/pages/notifications.cjsx

```diff
@@ -0,0 +1,145 @@
+React = require 'react'
+Loading = require '../components/loading-indicator'
+Paginator = require '../talk/lib/paginator'
+ChangeListener = require '../components/change-listener'
+talkClient = require '../api/talk'
+Notification = require './notifications/notification'
+
+module?.exports = React.createClass
+  displayName: 'NotificationsPage'
+
+  propTypes:
+    project: React.PropTypes.object
+    user: React.PropTypes.object
+
+  getDefaultProps: ->
+    query: page: 1
+
+  getInitialState: ->
+    firstMeta: { }
+    lastMeta: { }
+    notificationsMap: { }
+
+  componentWillMount: ->
+    @getNotifications() if @props.user
+
+  componentWillUnmount: ->
+    @markAsRead('first')()
+    @markAsRead('last')()
+
+  componentWillReceiveProps: (nextProps) ->
+    pageChanged = nextProps.query.page isnt @props.query.page
+    userChanged = nextProps.user and nextProps.user isnt @props.user
+    @getNotifications(nextProps.query.page) if pageChanged or userChanged
+
+  getNotifications: (page = @props.query.page) ->
+    page or= 1
+    query = {page}
+    query.section = "project-#{ @props.project.id }" if @props.project
+    query.section = @props.params.section if @props.params.section
+    talkClient.type('notifications').get(query).then (newNotifications) =>
+      meta = newNotifications[0]?.getMeta() or { }
+      notifications = @state.notifications or newNotifications
+      meta.notificationIds = (n.id for n in newNotifications)
+      notificationsMap = @state.notificationsMap
+
+      for notification in newNotifications
+        notificationsMap[notification.id] = notification
+
+      {firstMeta, lastMeta} = @state
+
+      if meta.page > @state.lastMeta.page
+        lastMeta = meta
+        notifications.push newNotifications...
+      else if meta.page < @state.firstMeta.page
+        firstMeta = meta
+        notifications.unshift newNotifications...
+      else
+        firstMeta = lastMeta = meta
+
+      @setState {notifications, notificationsMap, firstMeta, lastMeta}
+
+  markAsRead: (meta) ->
+    =>
```

Comment:
If the function was
``` coffeescript
markAsRead: (meta) ->
    do (meta) =>
      alert 'called'
```

It would run on page load since `do` is immediately calling the defined function to create the closure.